src/Symfony/Bridge/Doctrine/Middleware/Debug/DebugDataHolder.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Middleware\Debug;
+
+/**
+ * @author Laurent VOULLEMIER <laurent.voullemier@gmail.com>
+ */
+class DebugDataHolder
+{
+    private array $data = [];
+
+    public function addQuery(string $connectionName, Query $query): void
+    {
+        $this->data[$connectionName][] = [
+            'sql' => $query->getSql(),
+            'params' => $query->getParams(),
+            'types' => $query->getTypes(),
+            'executionMS' => static fn () => $query->getDuration(),  // stop() may not be called at this point
```

Comment:
```suggestion
            'executionMS' => $query->getDuration(...),  // stop() may not be called at this point
```

Reply:
not to be done if we want to target 5.4 instead.